tools/python/runfiles/runfiles.py

```diff
@@ -58,9 +58,41 @@
       p = subprocess.Popen([r.Rlocation("path/to/binary")], env, ...)
 """
 
+import inspect
 import os
 import posixpath
+import sys
 
+from .runfiles_constants import MAIN_REPOSITORY_RUNFILES_DIRECTORY
+
+
+def CurrentRepository(frame = 1):
+  """Returns the canonical name of the caller's Bazel repository.
+
+  Args:
+    frame: int; the stack frame to return the repository name for. Defaults to
+     1, the caller of the CurrentRepository function.
+  Returns:
+    the canonical name of the Bazel repository containing the Python file that
+    calls this function
+  Raises:
+    ValueError: if the caller cannot be determined or the caller's file path
+     is not contained in the Python runfiles tree
+  """
+  caller_path = inspect.getsourcefile(sys._getframe(frame))
```

Comment:
There may not be any source file, so this isn't a reliable way to try and find the runfiles.